bitswap/src/protocol.rs

```diff
@@ -71,7 +71,7 @@ where
     fn upgrade_outbound(self, mut socket: TSocket, _info: Self::Info) -> Self::Future {
         Box::pin(async move {
             let bytes = self.to_bytes();
-            upgrade::write_one(&mut socket, bytes).await
+            upgrade::write_length_prefixed(&mut socket, bytes).await
```

Comment:
Note while `write_one` closes the socket, `write_length_prefixed` does not.

Reply:
This one I did not notice, many thanks!

Reply:
Fixed in https://github.com/rs-ipfs/rust-ipfs/pull/472/commits/87aaefd616d21316146ddd8675a925104790b7ad